### PR TITLE
[feat] Add dbt osmosis utility to hub

### PR DESCRIPTION
**Pull request by z3z1ma**
Adds the dbt osmosis utility to meltano hub.

## Checklist

- [x] Add/update the file in the appropriate folder (`/taps` or `/targets`). The name of the file should match the name of the tap. If there is already one, add a descriptor to the name such as `-search`.
- [x] Add/update the PNG logo image in `/assets/logos/<taps or targets>`. The image name must match the YAML file name.
- [x] Tag `@tayloramurphy` or `@pnadolny13` to flag it for review. Or post to the [#hub](https://meltano.slack.com/archives/C01UGBSJNG5) channel on Meltano slack.

## Reviewer Checklist

- [x] Validate file against JSON Schema. https://www.jsonschema.net/home is an option.
- [x] Build website locally and validate everything works.

### CODE REVIEW

**Comment by z3z1ma** [comment]
@tayloramurphy @pnadolny13 

Up for review!

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+label: dbt-osmosis
+namespace: dbt_osmosis
+pip_url: dbt-osmosis
+maintenance_status: active
+repo: https://github.com/z3z1ma/dbt-osmosis
+executable: dbt-osmosis
+logo_url: /assets/logos/utilities/dbt-osmosis.png
+variant: z3z1ma
+default: true
+definition: A tool to supercharge dbt development by providing 3 key functions.
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:11]:
```suggestion
definition: |
   is a tool to supercharge [dbt](https://docs.getdbt.com/) development by providing 3 key functions:
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  1. A workbench for a REPL like development experience for dbt models
+  2. The ability to diff model outputs across file revisions,
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:13]:
```suggestion
  2. The ability to diff model outputs across file revisions
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  3. Automation which manages generating, updating, and propagate documentation for your
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:14]:
```suggestion
  3. Automation which manages generating, updating, and propagating documentation for your
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  schema yaml files.
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:15]:
```suggestion
  `schema.yml` files.
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  Additionally, a server for VS code to allow quickly previewing models
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:16]:
```suggestion
  Additionally, a server for VS code enables quickly previewing models
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  when used in conjunction with https://github.com/datacoves/vscode-dbt-power-user
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:17]:
```suggestion
  when used in conjunction with the [VSCode dbt power user extension](https://github.com/datacoves/vscode-dbt-power-user).
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  though the server can be used for anything.
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:18]:
```suggestion
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  compose:
+    args: compose --project-dir ${MELTANO_PROJECT_ROOT}/transform --profiles-dir
+      ${MELTANO_PROJECT_ROOT}/transform/profiles/${ADAPTER}
+    description: A slimmer version of `run` used in precommit hooks
+settings:
+  - name: adapter
+    env: ADAPTER
+    description: |
+      This should be set to the adapter you have installed as meltano will install
+      dbt and its profile directory into transform/&lt;adapter&gt; directory. We use this
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:61]:
```suggestion
      dbt and its profile directory into `transform/<adapter>` directory. We use this 
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    description: |
+      This should be set to the adapter you have installed as meltano will install
+      dbt and its profile directory into transform/&lt;adapter&gt; directory. We use this 
+      config option to resolve dbt profile directory.
+keywords:
+  - dbt
+  - osmosis
+  - tooling
+next_steps: |
+  1. Add dbt-&lt;adapter&gt; to your pip_url, ie: `pip_url: dbt-osmosis dbt-bigquery`
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:68]:
```suggestion
  1. Add dbt-<adapter> to your pip_url, ie: `pip_url: dbt-osmosis dbt-bigquery`
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+  2. Override the namespace to mirror your dbt adapter, ie: `namespace: dbt_bigquery`
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:70]:
```suggestion
  1. Override the namespace to mirror your dbt adapter, ie: `namespace: dbt_bigquery`
```

File: _data/meltano/utilities/dbt-osmosis/z3z1ma.yml [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+  3. Mirror the configuration set in your dbt-&lt;adapter&gt; plugin. A simple copy-paste of
```
Comment (tayloramurphy) [inline _data/meltano/utilities/dbt-osmosis/z3z1ma.yml:72]:
```suggestion
  1. Mirror the configuration set in your dbt-<adapter> plugin. A simple copy-paste of
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]
Just a few text changes, but it's looking good! Preview is here https://deploy-preview-766--meltano-hub.netlify.app/utilities/dbt-osmosis

**Review by tayloramurphy — APPROVED** [review]
This looks good to me. @pnadolny13 can you review and merge if you approve?

**Comment by pnadolny13** [comment]
@z3z1ma One last thing you just need to move your logo because the `assets/logos/..` directory was migrated into `static/assets/logos/..`. Then I should be able to test an add from my project just to verify its working as expected.

**Comment by z3z1ma** [comment]
@pnadolny13 

Got it, all done -- give it a whirl!

**Pull request merged**